src/stdio_mgr/stdio_mgr.py

```diff
@@ -28,28 +28,32 @@
 
 import sys
 from contextlib import contextmanager
-from io import StringIO, TextIOBase
+from io import BufferedReader, BytesIO, StringIO, TextIOBase, TextIOWrapper
 
 import attr
 
 
 @attr.s(slots=False)
-class TeeStdin(StringIO):
+class TeeStdin(TextIOWrapper):
     """Class to tee contents to a side buffer on read.
 
-    Subclass of :cls:`~io.StringIO` that overrides
-    :meth:`~io.StringIO.read` and :meth:`~io.StringIO.readline`
+    Subclass of :cls:`~io.TextIOWrapper` that overrides
+    :meth:`~io.TextIOWrapper.read` and :meth:`~io.TextIOWrapper.readline`
     to tee all content *read* from the stream to `tee`. The
     canonical use-case is with :func:`stdio_mgr`,
     where `tee` is the mocked stream for `stdin`.
 
     To emphasize: teeing occurs on content *read*, **not write**.
 
-    This class also provides the method
-    :meth:`TeeStdin.append`, which is not available
-    for the base :cls:`~io.StringIO` type.
-    This method adds new content to the end of the
-    stream while leaving the read position unchanged.
+    This class also provides two additional methods
```

Comment:
@jayvdb, please review this docstring change for accuracy.

Reply:
sure

Reply:
<nod>, no problem.

You could just squash-rebase the two commits, if you're ok with the language as-is. Otherwise, sure, just reset to & amend the prior commit.

Reply:
I provided the same in my own words, and expanded it a bit, without so many modifications to your prior text.  There are more changes afoot, and it would be good if you outlined your style preferences in an issue, or baked them into CI, when you get a chance.

Reply:
<nod>, I've already put flake8+plugins into CI on another project; the plan is definitely to roll that machinery over to this and my other projects. I'll prioritize it.